packages/ui-components/src/chat/message.tsx

```diff
@@ -0,0 +1,176 @@
+    box-shadow: ${(props) => props.theme.shadow.medium};
+`;
+
+const MessageWrapper = styled.div`
+    position: relative;
+
+    gap: 0.5rem;
+
+    width: 100%;
+    padding: 1rem;
+
+    background-color: ${(props) => props.theme.colors.blue1};
+    border-radius: 0.25rem;
+    box-shadow: ${(props) => props.theme.shadow.medium};
+
+    :hover ${InteractiveIcons} {
+        display: flex;
+    }
+`;
+
+const MessageTop = styled.div`
+    display: flex;
+    justify-content: space-between;
+
+    width: 100%;
+    height: 1.625rem;
+
+    font-size: 0.8rem;
+    color: ${(props) => props.theme.colors.grey5};
+`;
+
+const MessageBody = styled.div`
+    display: flex;
+    width: 100%;
+    color: ${(props) => props.theme.colors.text};
+`;
+
+const DeleteIcon = styled(Trash)`
+    height: 0.8rem;
+    color: ${(props) => props.theme.colors.secondary};
+
+    :hover {
+        color: ${(props) => props.theme.colors.secondaryHover}CC;
+    }
+
+    :active {
+        color: ${(props) => props.theme.colors.secondaryDown}99;
+    }
+`;
+
+const EditIcon = styled(PenToSquare)`
+    height: 0.8rem;
+    color: ${(props) => props.theme.colors.secondary};
+
+    :hover {
+        color: ${(props) => props.theme.colors.secondaryHover}CC;
+    }
+
+    :active {
+        color: ${(props) => props.theme.colors.secondaryDown}99;
+    }
+`;
+
+const EditButtons = styled.div`
+    display: flex;
+    gap: 1rem;
+    justify-content: flex-end;
+`;
+
+export interface MessageProps extends InteractiveComponentProps<Message> {
+    /** An optional onChange handler for listening to changes in the input */
+    onChange?: (value: Message, e?: React.SyntheticEvent<HTMLInputElement>) => void | Promise<void>;
+    /** An optional event listener for complete events (enter presses) */
+    onDelete?: (id: string) => void | Promise<void>;
+}
+
+/**
+ * A Message component that displays a message with a timestamp and allows for editing and deleting
+ *
+ * @param {MessageProps} props - the component props
+ */
+function MessageComponent(props: MessageProps): JSX.Element {
+    const [editMode, setEditMode] = React.useState(false);
+    const [value, setValue] = React.useState(props.value);
+    const [editMessage, setEditMessage] = React.useState(props.value.message);
+
+    const onAccept = (): void => {
+        // if the message hasn't changed, don't do anything
+        if (editMessage === value.message) {
+            return;
+        }
+        const newMessage = { ...value, message: editMessage };
+        if (props.onChange) {
+            props.onChange(newMessage);
+        } else {
+            // Update internal state only if it's uncontrolled
+            setValue(newMessage);
+        }
+        setEditMode(false);
+    };
+
+    const onDelete = (): void => {
+        if (props.onDelete) {
+            props.onDelete(props.value.id);
+        }
+    };
+
+    React.useEffect(() => {
+        setValue(props.value);
+    }, [props.value]);
```

Comment:
Same as in the other component this could just be an inline if block with setState in it

Reply:
Oops, forgot about that one, fixed now